pymc3/distributions/multivariate.py

```diff
@@ -503,17 +503,20 @@ def __init__(self, n, p, *args, **kwargs):
         self.mode = tt.cast(tround(self.mean), 'int32')
 
     def _random(self, n, p, size=None):
+        # Set float type to float64 for numpy.
+        p = p.astype('float64')
+        # Now, re-normalize all of the values in float64 precision.
         if size == p.shape:
             size = None
-
         if p.ndim == 1:
+            p = p / (p.sum())
             randnum = np.random.multinomial(n, p.squeeze(), size=size)
         elif p.ndim == 2:
+            p = p / (p.sum(axis=1, keepdims=True))
             randnum = np.asarray([np.random.multinomial(n, pp, size=size) for pp in p])
         else:
             raise ValueError('Outcome probabilities must be 1- or 2-dimensional '
                              '(supplied `p` has {} dimensions)'.format(p.ndim))
-
         return randnum
```

Comment:
should the return value get cast back to the original data type?  honest question -- maybe @kyleabeauchamp has an opinion?